CMakeLists.txt

```diff
@@ -34,7 +34,8 @@ IRODS_MACRO_CHECK_DEPENDENCY_SET_FULLPATH_ADD_TO_IRODS_DEVELOP_DEPENDENCIES_LIST
 string(REPLACE ";" ", " IRODS_PACKAGE_DEPENDENCIES_STRING "${IRODS_PACKAGE_DEPENDENCIES_LIST}")
 string(REPLACE ";" ", " IRODS_DEVELOP_DEPENDENCIES_STRING "${IRODS_DEVELOP_DEPENDENCIES_LIST}")
 
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
should we disable compiler extensions?

what do we gain/lose by doing this?

Reply:
@alanking @SwooshyCueb any thoughts on this?

Reply:
Disabling this feels like the right approach until we find that we need some extension.

The documentation states...

> Note that because most compilers enable extensions by default, this may expose portability bugs in user code or in the headers of third-party dependencies.

Found here: https://cmake.org/cmake/help/latest/manual/cmake-compile-features.7.html#availability-of-compiler-extensions

But I'm probably not the most trustworthy authority for that question :)

Reply:
i had the same thoughts around that.

unless @SwooshyCueb has any thoughts on it, i'm leaning towards disabling extensions. if we need them, we can re-enable them.

Reply:
marking as resolved.

if we need extensions, we can re-enable them in a later commit.

Reply:
Apologies, this didin't show up in my notifications for some reason.

I agree with disabling extensions if we don't need them or have a specific reason to keep them enabled.
This isn't something I would typically do, myself, but I don't fault your logic here and I ultimately think it's the right choice for the project.